### PR TITLE
Added functionality to calculate difference between best price of month/year and current day

### PR DESCRIPTION
**Pull request by Cebix90**
Added functionality to calculate difference between best price of month/year and current day with tests

### CODE REVIEW

File: src/main/java/org/currencygoldexchangeapp/constants/ListOfHolidays.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                LocalDate.of(year, 5, 30),
+                LocalDate.of(year, 8, 15),
+                LocalDate.of(year, 11, 1),
+                LocalDate.of(year, 11, 11),
+                LocalDate.of(year, 12, 25),
+                LocalDate.of(year, 12, 26)
+        );
+    }
+
+    public boolean contains(LocalDate date) {
```
Comment (blazejprzyluski) [inline src/main/java/org/currencygoldexchangeapp/constants/ListOfHolidays.java]: contains what?

File: src/main/java/org/currencygoldexchangeapp/constants/ListOfHolidays.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package org.currencygoldexchangeapp.constants;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+public class ListOfHolidays {
+    public final List<LocalDate> holidays;
```
Comment (blazejprzyluski) [inline src/main/java/org/currencygoldexchangeapp/constants/ListOfHolidays.java]: why public?

File: src/main/java/org/currencygoldexchangeapp/exceptions/ExceededResultsLimitException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package org.currencygoldexchangeapp.exceptions;
+
+public class ExceededResultsLimitException extends RuntimeException {
+    public ExceededResultsLimitException() {
+        super("Data series has been exceeded");
```
Comment (blazejprzyluski) [inline src/main/java/org/currencygoldexchangeapp/exceptions/ExceededResultsLimitException.java]: Please make it more descriptive

File: src/test/java/GoldValueAPIHandlerTest.java [hunk 4]
```diff
@@ -29,182 +34,343 @@ public class GoldValueAPIHandlerTest {
     private HttpResponse<String> response;
 
     @InjectMocks
-    private GoldValueAPIHandler handler;
+    private GoldValueAPIHandler goldValueAPIHandler;
+
+    @Nested
+    class TestGetGoldValueForSpecificDate {
+        @Test
+        public void returnsCorrectGoldValue() {
```
Comment (blazejprzyluski) [inline src/test/java/GoldValueAPIHandlerTest.java:42]: Let's discuss test naming on 1:1

File: src/main/java/org/currencygoldexchangeapp/constants/ListOfHolidays.java (hunk 2 above)
Comment (Cebix90) [reply src/main/java/org/currencygoldexchangeapp/constants/ListOfHolidays.java]: true, it should be private

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**